src/Deinterleave.cpp

```diff
@@ -176,224 +178,637 @@ Stmt collect_strided_stores(const Stmt &stmt, const std::string &name, int strid
     return collect.mutate(stmt);
 }
 
+// Essentially, broadcast the mask, i.e. take each mask element, and repeat it
+// \p scale times, consequtively. No reordering of mask elements happens!
+std::vector<char> upscale_mask(const std::vector<char> &src_mask, int scale) {
+    internal_assert(scale > 1) << "Expected to at least double the mask.";
+    std::vector<char> mask;
+    mask.reserve(scale * src_mask.size());
+    for (char src_mask_elt : src_mask) {
+        std::fill_n(std::back_inserter(mask), /*count=*/scale,
+                    /*value=*/src_mask_elt);
+    }
+    return mask;
+}
+
+// Compute the elements of the specified ramp/linear space.
+std::vector<int> ramp_to_indexes(int starting_lane, int lane_stride,
```

Comment:
Change indexes to indices. The singular is "index" and the plural is "indices" because for some god damn reason English sometimes uses latin plurals and it's a 3rd declension noun and why oh why did I take four years of Latin.

Reply:
Oh, index, but indices, got it.

Reply:
<A "well, actually..." emerges>

[Both "indexes" and "indices" are acceptable English usages](https://www.nasdaq.com/articles/indexes-or-indices-whats-the-deal-2016-05-12)

Reply:
> Though both are still widely used, they take on different usage in their senses. "Indices" is used when referring to mathematical, scientific and statistical contexts.